src/dmadata/dmadata.c

```diff
@@ -0,0 +1,30 @@
+#include "z64dma.h"
+
+// Linker symbol declarations (used in the table below)
+#define DEFINE_DMA_ENTRY(name) \
+    extern u8 _##name##SegmentRomStart[]; \
+    extern u8 _##name##SegmentRomEnd[];
+
+#define DEFINE_DMA_ENTRY_END()
+
+#include "tables/dmadata_table.h"
+
+#undef DEFINE_DMA_ENTRY
+#undef DEFINE_DMA_ENTRY_END
+
+// dmadata Table definition
+#define DEFINE_DMA_ENTRY(name) \
+    { (u32)_##name##SegmentRomStart, (u32)_##name##SegmentRomEnd, (u32)_##name##SegmentRomStart, 0 },
+
+#define DEFINE_DMA_ENTRY_END() \
+    { 0 },
+
+DmaEntry gDmaDataTable[] = {
+    #include "tables/dmadata_table.h"
```

Comment:
Iirc the formatter removes the indentation for these (which is fine, it just means the files likely need a format run)